stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt

```diff
@@ -0,0 +1,130 @@
+package io.getstream.chat.ui.sample.feature.chat.info.group.member
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.QuerySort
+import io.getstream.chat.android.client.call.await
+import io.getstream.chat.android.client.channel.ChannelClient
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.client.utils.FilterObject
+import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.livedata.controller.QueryChannelsController
+import io.getstream.chat.android.livedata.utils.Event
+import io.getstream.chat.ui.sample.common.isDraft
+import kotlinx.coroutines.launch
+
+class GroupChatInfoMemberOptionsViewModel(
+    cid: String,
+    private val memberId: String,
+    private val chatDomain: ChatDomain = ChatDomain.instance(),
+    chatClient: ChatClient = ChatClient.instance(),
+) : ViewModel() {
+
+    private val channelClient: ChannelClient = chatClient.channel(cid)
+    private val _events = MutableLiveData<Event<UiEvent>>()
+    private val _state: MediatorLiveData<State> = MediatorLiveData()
+    val events: LiveData<Event<UiEvent>> = _events
+    val state: LiveData<State> = _state
+
+    init {
+        viewModelScope.launch {
+            val result = chatDomain.useCases.queryChannels(
+                filter = Filters.and(
+                    Filters.eq("type", "messaging"),
+                    FilterObject(
+                        data = mutableMapOf(
+                            "distinct" to true,
+                            "members" to listOf(memberId, chatDomain.currentUser.id),
+                        ),
+                    ),
+                ),
+                sort = QuerySort.desc("last_updated"),
+                messageLimit = 0,
+                limit = 1,
+            ).await()
+            if (result.isSuccess) {
+                _state.addSource(
+                    Transformations.map(result.data().channelsState) { channelState ->
```

Comment:
What if for better redabillity move inner lambda to some method? E.g. `fun mapChannelState(channelState: ChannelState): State`?